mozjs-sys/Cargo.toml

```
@@ -35,7 +35,8 @@ libz-sys = "1.1.19"
 encoding_c = "0.9.8"
 encoding_c_mem = "0.2.6"
 # unicode-bidi-ffi = { path = "./mozjs/intl/bidi/rust/unicode-bidi-ffi" }
-icu_capi = "=1.5.0" # keep in sync with intl/icu_capi/Cargo.toml
+# keep in sync with intl/icu_capi/Cargo.toml
+icu_capi = {  version = "=1.5.0", default-features = false, features = ["compiled_data", "icu_segmenter"] }
```

```suggestion
icu_capi = { version = "=1.5.0", default-features = false, features = ["compiled_data", "icu_segmenter"] }
```

How did you determine the features needed? Because from what I see firefox keeps all default (`["compiled_data", "default_components", "logging", "simple_logger", "std"]`).

---

https://github.com/mozilla-firefox/firefox/blob/esr128/intl/icu4x-patches/001-Cargo.toml.patch

Firefox applies a patch to reduce the `compiled_data`, and we tried if we can achieve the same by just removing components. We need `icu_segmenter` and the data it references, and that seems to be sufficient for mozjs to compile and servo to pass all wpt tests (linked a try run in the companion PR). 
Do you think keeping the logging feature might be valuable?

---

No, but I think this might be reason for compilation problems. I suspect we need at least std and all features enabled here: https://github.com/servo/mozjs/blob/0bebd6872b1a061b69bcfcf73162abf0af27eebc/mozjs-sys/mozjs/intl/icu_capi/Cargo.toml#L192 (this is from SM 137 not 128).

---

```suggestion
icu_capi = { version = "=1.5.0", default-features = false, features = ["std", "compiled_data", "icu_calendar", "icu_properties", "icu_segmenter"] }
```

---

Downgrading the compiler does fix the problem on windows. Looking at the undefined functions, it seems that the problem should be fixed by linking`Advapi32.dll`. That file does appear in the [mozglue/moz.build](https://github.com/servo/mozjs/blob/0bebd6872b1a061b69bcfcf73162abf0af27eebc/mozjs-sys/mozjs/mozglue/build/moz.build#L35), but I guess the linker is invoked via rust, so we need to add a line to our build-script.

---

That's a good point. I'll add them.

---

Wouldn't it still cause link issues when building the mozjs tests, if we didn't add a required feature?